Add competitive orders (#14)
robocup-logistics/rcll-rulebook: rulebook.tex

requiring 1 base, and two colors requiring no additional base at
all. Products retrieved from the Storage Station may be used only for
$C_0$ orders with a requested quantity equal to or greater than two.
An order can be \emph{competitive} or \emph{non-competitive}. For a
competitive order, only the team that delivers first will score points

> for the delivery, and it will score twice the amount of points. The

Author: But on the other hand, if the team scores the same amount of points, the incentive to pursue a competitive order is rather low, because it is more risky and offers no benefit other than blocking the other team from scoring those points. This may be sufficient if there are a number of competitive orders, but if we limit them, then the other team may also just decide to ignore it.

I think pursuing a competitive order should have a benefit other than blocking the other team from scoring.

> for the delivery, and it will score twice the amount of points. The

Reviewer: The incentive is to "steal" the points from the other team. But I do see your point that at the current state some (most?) teams will simply pursue their guaranteed orders.

I think what I really dislike is a "factor" of points. We do not have this anywhere. If we keep this, let's make this a static number of points, e.g., 10 points on successful (accepted) competitive delivery. In the refbox, there is currently no notion of factors, just summed points. I don't think we need to account for late delivery for the competitive points addition, if you deliver and score at all, you get 10 points (or so) extra.

> for the delivery, and it will score twice the amount of points. The

Author: The idea of the factor was that it makes the rule independent of the actual scoring scheme for the non-competitive variant. But I think we can also rephrase this as only giving additional points for the delivery of a competitive order, and no points for delivery if it is the second delivery for that order.

This would reduce the penalty for a second delivery for a competitive complex order, as the team would still get points for the intermediate steps, just not for the delivery anymore.

> for the delivery, and it will score twice the amount of points. The

Author: We could also remove the time limit on the competitive order, i.e., make it a standing order. This would also fit with the real-world scenario in mind, where the manufacturer just wants to get the product as fast as possible, but without a specified deadline. What do you think?

This would make the scoring a lot simpler.

> for the delivery, and it will score twice the amount of points. The

Author: I got rid of the factor and assigned +20 for the first delivery, which doubles the delivery points for a regular delivery.

> for the delivery, and it will score twice the amount of points. The

Reviewer: This is too much in my opinion. It is given too much emphasis on the competitive order. As I stated earlier, let's start slowly and see how it goes. 10 seems to be enough for me, that's ~30% of the overall product you get on top in addition of the fact that a competing team might just have lost work effort worth 30 points if they were slower.

Making it a standalone order makes it less of a strategic run-time decision, I'd keep it with a time window (in overtime the time window would span the entire overtime).

> for the delivery, and it will score twice the amount of points. The

Author: I reduced the bonus to 10 points.

other team will not get any points for the delivery, even if it delivers
in time.  For a non-competitive order, both teams score on delivery with
a factor of $1$.

In each game, up to 9 orders will be placed within the regular game.

> 4 of those 9 orders will be \emph{competitive}, while the remaining

Author: I'm changing it to 1 + 1 in overtime.

orders will be \emph{non-competitive}. In case of overtime, an
additional competitive order will be placed.\footnote{The numbers can
be adjusted during the tournament by the Technical Committee, should
this turn out to be necessary.}
The complexities of orders will be similar for all games within a

> tournament phase. But the actual production chains are randomized.

Reviewer: What has changed here? Is it just reformatting? In that case, remove. Otherwise, please give a hint.

Author: I added the sentence
> A competitive order always requires exactly one product to be delivered.

requiring 1 base, and two colors requiring no additional base at
all. Products retrieved from the Storage Station may be used only for
$C_0$ orders with a requested quantity equal to or greater than two.
An order can be \emph{competitive} or \emph{non-competitive}. For a
competitive order, only the team that delivers first will score points

> for the delivery, and it will score twice the amount of points. The

Reviewer: ```suggestion
for the delivery, and it will score higher (cf.~\reftab{tab:scoring-production}). The
```
This is wrong. Delivery in time 20 + 1st competitive 10 = 30, which is not twice the amount of points. This should just state "scores higher" and leave the actual value to the table, as to avoid inconsistencies if there are multiple sources of information for the same thing (we had plenty of those in the early years).

Author: Thanks for spotting this :+1:

awarded. Performing a step for a later order which has not yet been
announced cannot be awarded.

For a competitive order, the same scoring scheme for in-time, delayed,

> late, and wrong delivery applies, with an additional factor of $2$ if

Reviewer: Again, no factor in my opinion.

        delivered
        & $+1$\\
        False delivery & Deliver an intermediate product & $0$\\

>         1st competitive delivery & Factor for delivering a product for a competitive

Reviewer: Can be removed if my comment from above is accepted.

awarded. Performing a step for a later order which has not yet been
announced cannot be awarded.

> For a competitive order, the same scoring scheme for in-time, delayed, and late

Reviewer: What's the difference between delayed and late delivery?

Author: Delayed: within 10 seconds after the deadline, late: more than 10 seconds after the deadline (as specified in the scoring table)

Reviewer: Ah, got ya.

delivery applies, with additional points for a successful first delivery
and a point deduction for the second delivery (cf.~\reftab{tab:scoring}). The

> deduction may not exceed the points given for the delivery.\footnote{As an

Reviewer: Place the "." after the footnote. We use the fnpct package which makes the display nicer, but it requires points to be set after the footnote (in LaTeX you seemingly can only do lookahead, but not look back).
```suggestion
deduction may not exceed the points given for the delivery\footnote{As an
```

  example, if a team delivers late and scores only $5$ points for the delivery,

> only $5$ points will be deducted.} Points for production steps are given in the

Reviewer: ```suggestion
only $5$ points will be deducted.}. Points for production steps are given in the
```

Author: Fixed, thanks for the suggestion

awarded. Performing a step for a later order which has not yet been
announced cannot be awarded.

For a competitive order, the same scoring scheme for in-time, delayed, and late

> delivery applies for the first delivery, with an addition of $10$ points for a

Reviewer: This explicit score should be removed here, point to the table, reduce redundancies especially for the crucial scoring bit (we ran into ambiguities in the past if only one place was updated and the other forgotten).

Author: Makes sense. I did not change the sentence about giving no points to the second delivery, as this will most probably not change, and it helps to understand competitive orders.

        \\
        False delivery & Deliver an intermediate product & $0$
        \\
        1st competitive delivery
        & Points for the first delivery for a competitive order. The score is
        given in addition to the points for the regular delivery.
        &$+ 10$
        \\
        2nd competitive delivery
        & Point deduction for the second delivery for a competitive order. The

>         points are deducted from the delivery points.

Reviewer: Here would be a good place to state once more that zero is the minimum. This is what teams may look at for scores more frequently.

Author: Yes, makes sense, added.
